### PR TITLE
Add “See also” links to favorite class API docs

### LINKED ISSUE #62365: Add cross-links from api.flutter.dev to flutter.dev/docs
From recent surveys, we have good evidence that Flutter developers are struggling with topics that are well explained on [flutter.dev/docs](https://flutter.dev/docs). In other words, we have the content that could help, but Flutter developers aren't finding it.

We currently have 280 pages on flutter.dev/docs. Nobody reads them top to bottom. Developers just don't know about all the nuggets of information that's in there, and they don't necessarily know or remember to search for it. (Googlers: http://go/thickening-tail.)

Primary research also suggests that "the % of users who visited tutorial/codelab and then API docs (or vice versa) was almost negligible. They started clicking on the links on the page and stayed in the API doc loop or website loop." (@jayoung-lee) And we have good indication from secondary research that cross linking helps users find what they need. (@cobblest)

## Use case

For example, if a new Flutter developer finds her way to the [`Simulation`](https://api.flutter.dev/flutter/physics/Simulation-class.html) class on api.flutter.dev, she should be presented with a "See also" link that leads to the [Animate a widget using a physics simulation](https://flutter.dev/docs/cookbook/animation/physics-simulation) cookbook recipe on flutter.dev/docs.

This is already happening in some cases. See, for example, the last line in the [`TextField`](https://api.flutter.dev/flutter/material/TextField-class.html) dartdoc.

## Proposal

I'm proposing to take the top 20-100 most-visited api.flutter.dev classes, and see if I can add meaningful links from them to flutter.dev/docs pages, through dartdocs.

I'm proposing to do this work myself, in a single PR. I just want to see if this is in line with what everyone's expecting from API docs.

### PR DESCRIPTION
**Pull request by filiph**
This change takes the top 20 most popular API docs (according to analytics) and cross-links them to relevant pages on flutter.dev, especially Cookbook pages.

Fixes https://github.com/flutter/flutter/issues/62365. Though there’s definitely more we can do — we have ~20,000 pages on api.flutter.dev.

## Description

This is a dartdoc page.

## Related Issues

* [Add cross-links from api.flutter.dev to flutter.dev/docs #62365](https://github.com/flutter/flutter/issues/62365)

## Tests

I added no tests: doc change.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/app_bar.dart [hunk 1]
```diff
@@ -165,13 +165,19 @@ class _ToolbarContainerLayout extends SingleChildLayoutDelegate {
 ///  * [Scaffold], which displays the [AppBar] in its [Scaffold.appBar] slot.
 ///  * [SliverAppBar], which uses [AppBar] to provide a flexible app bar that
 ///    can be used in a [CustomScrollView].
+///    * Cookbook: [Place a floating app bar above a list](https://flutter.dev/docs/cookbook/lists/floating-app-bar)
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/app_bar.dart:168]: nit: indentation.

File: packages/flutter/lib/src/widgets/container.dart [hunk 2]
```diff
@@ -248,6 +248,7 @@ class DecoratedBox extends SingleChildRenderObjectWidget {
 ///
 ///  * [AnimatedContainer], a variant that smoothly animates the properties when
 ///    they change.
+///    * Cookbook: [Animate the properties of a container](https://flutter.dev/docs/cookbook/animation/animated-container)
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/container.dart:251]: nit: indentation

**Review by goderbauer — APPROVED** [review]
LGTM once cirrus is happy

File: packages/flutter/lib/src/material/app_bar.dart (hunk 1 above)
Comment (filiph) [reply packages/flutter/lib/src/material/app_bar.dart:168]: This is intended. The cookbook article talks about floating bar, so I added it as a sub-node to the SliverAppBar list item. If that's too confusing or if we try to stay away from nested lists in API docs, I can correct this (and the other instance below).

Comment (goderbauer) [reply packages/flutter/lib/src/material/app_bar.dart:168]: I would prefer not nesting these to keep them simpler.

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by goderbauer** [comment]
The analyzer is also complaining about some trailing whitespace.

**Comment by filiph** [comment]
Feedback addressed.

I also completely botched the rebase on my side, which I think is the reason a bot auto-added owners (@christopherfujino, @jmagman, @Piinks).  I removed y'all because I think the addition was in error, but feel free to give me feedback.

2 web tests timed out. As this is a docs-only change, I'm going to ignore that, and I'm going to attach the "waiting for green" label. Please tell me if that's a mistake.

**Comment by christopherfujino** [comment]
> This pull request is not suitable for automatic merging in its current state.
> 
> * The status or check suite web_tests-2-linux has failed. Please fix the issues identified (or deflake) before re-applying this label.
> * The status or check suite web_tests-11_last-linux has failed. Please fix the issues identified (or deflake) before re-applying this label.

Our new machine overlords are not happy.

**Comment by christopherfujino** [comment]
> Feedback addressed.
> 
> I also completely botched the rebase on my side, which I think is the reason a bot auto-added owners (@christopherfujino, @jmagman, @Piinks). I removed y'all because I think the addition was in error, but feel free to give me feedback.
> 
> 2 web tests timed out. As this is a docs-only change, I'm going to ignore that, and I'm going to attach the "waiting for green" label. Please tell me if that's a mistake.

The bot won't merge unless it's all green. Re-running the timeouts.

**Comment by jmagman** [comment]
Also won't automerge until flutter-gold is successful.  Looks like that kicked off 4 minutes ago.

**Comment by filiph** [comment]
Thanks both! I'll try again now. Looks like the only missing thing is `flutter-build`.

**Pull request merged**